sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+import pytest
+from sklearn.base import RegressorMixin, ClassifierMixin, BaseEstimator
+from sklearn.multioutput import MultiOutputRegressor, MultiOutputClassifier
+from sklearn.utils.metadata_routing import MetadataRouter
+from sklearn.tests.test_metadata_routing import (
+    record_metadata,
+    check_recorded_metadata,
+    assert_request_is_empty,
+)
+
+N, M = 100, 4
+X = np.random.rand(N, M)
+y = np.random.randint(0, 2, size=N)
+y_multi = np.random.randint(0, 2, size=(N, 3))
+metadata = np.random.randint(0, 10, size=N)
+sample_weight = np.random.rand(N)
+
+
+class ConsumingRegressor(RegressorMixin, BaseEstimator):
+    """A regressor consuming metadata."""
+
+    def __init__(self, **kwargs):
+        for param, value in kwargs.items():
+            setattr(self, param, value)
+
+    def partial_fit(self, X, y, sample_weight=None, metadata=None):
+        record_metadata(
+            self, "partial_fit", sample_weight=sample_weight, metadata=metadata
+        )
+        return self
+
+    def fit(self, X, y, sample_weight=None, metadata=None):
+        record_metadata(self, "fit", sample_weight=sample_weight, metadata=metadata)
+        return self
+
+    def predict(self, X, y, sample_weight=None, metadata=None):
+        record_metadata(self, "predict", sample_weight=sample_weight, metadata=metadata)
+        return np.zeros(shape=(len(X)))
+
+
+class ConsumingClassifier(ClassifierMixin, BaseEstimator):
```

Comment:
Those could also be next to the minimal (or inherited) from the basic/minimal classifiers/regressors.

Reply:
Maybe a small comment if you think this is appropriate.